external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/trident/KafkaTridentSpoutEmitter.java

```diff
@@ -97,13 +110,48 @@ public KafkaTridentSpoutEmitter(KafkaTridentSpoutConfig<K, V> kafkaSpoutConfig,
         this.firstPollOffsetStrategy = kafkaSpoutConfig.getFirstPollOffsetStrategy();
         this.startTimeStamp = kafkaSpoutConfig.getStartTimeStamp();
         LOG.debug("Created {}", this.toString());
+
+        registerMetric();
+    }
+
+    /**
+     * Acquires metric instances through registration with the TopologyContext.
+     */
+    private void registerMetric() {
+        LOG.info("Registering Spout Metrics");
+
+        String configGroupId = "";
+        if (kafkaSpoutConfig.getKafkaProps().get(ConsumerConfig.GROUP_ID_CONFIG) != null) {
+            configGroupId = kafkaSpoutConfig.getKafkaProps().get(ConsumerConfig.GROUP_ID_CONFIG).toString() + UNDERSCORE;
+        }
+
+        eventEmitRate = topologyContext.registerMeter(
+            configGroupId + EVENT_EMIT_METRIC_NAME);
+        kafkaClientMaxLag = topologyContext.registerGauge(
+            configGroupId + KAFKA_CLIENT_MAX_LAG_METRIC_NAME,
+            new Gauge<Double>() {
+                @Override
+                public Double getValue() {
+                    if (consumer == null) {
+                        return 0.0;
+                    }
+                    // Extract spout lag from consumer's internal metrics
+                    for (Map.Entry<MetricName, ? extends Metric> metricKeyVal : consumer.metrics().entrySet()) {
+                        Metric metric = metricKeyVal.getValue();
+                        if (metric.metricName().name().equals(INTERNAL_KAFKA_RECORDS_LAG_MAX_METRIC)) {
```

Comment:
If I understand correctly, there are actually two types of "records-lag-max" metrics, 

one is partition level 
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetcherMetricsRegistry.java#L133-L134

and the other one is consumer level
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetcherMetricsRegistry.java#L98-L99

They use the same name as "records-lag-max", while the tags are different. In this case, it is hard to tell what the first "records-lag-max" metric from `consumer.metrics()` is.

I noticed that older kafka version has different way of doing things . 
https://github.com/apache/kafka/blob/0.11.0/clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetcherMetricsRegistry.java#L108

We will to take care of compatibility issues here.